Test/comp/DowncastDt.dfy

```diff
@@ -8,22 +8,41 @@
 // Java avoids this problem by not even allowing traits in datatypes.
 
 datatype Co<+T> = Co(T)
+datatype Con<-T> = Con(T -> bool)
 
 trait X {}
 
 class Y extends X {
   constructor Y() {}
 }
 
-method Downcast() {
+method DowncastCo() {
   var i := new Y.Y();
   var a: Co<X> := Co(i);
   var b: Co<Y>;
   b := a;
   print a, " and ", b, "\n";
 }
 
+method DowncastCon() {
+  var i: Con<X> := Con(_ => false);
+  var a: Con<Y> := i;
+  var b: Con<X>;
+  b := a;
+  print a, " and ", b, "\n";
+}
+
+method DowncastFunc() {
+  var i := new Y.Y();
+  var a: bool -> X := (_ => i);
+  var b: bool -> Y;
+  b := a;
+  print a, " and ", b, "\n";
```

Comment:
Same here: Can you not print a and b, but instead perhaps a(false) and b(false)? That way you avoid testing the exact implicit names of function arguments.